routing/route.cpp

```diff
@@ -354,6 +359,42 @@ void Route::Update()
   m_currentTime = 0.0;
 }
 
+void Route::AppendTraffic(Route const & route)
+{
+  CHECK(route.IsValid(), ());
+
+  if (GetTraffic().empty() && route.GetTraffic().empty())
+    return;
+
+  if (!IsValid())
+  {
+    m_traffic.assign(route.GetTraffic().cbegin(), route.GetTraffic().cend());
+    return;
+  }
+
+  // Note. At this point the last item of |m_poly| should be removed.
+  // So the size of |m_traffic| should be equal to size of |m_poly.
+  if (GetTraffic().empty())
+  {
+    CHECK_GREATER_OR_EQUAL(m_poly.GetPolyline().GetSize(), 1, ());
+    m_traffic.resize(m_poly.GetPolyline().GetSize(), SpeedGroup::Unknown);
+  }
+
+  CHECK_EQUAL(GetTraffic().size(), m_poly.GetPolyline().GetSize(), ());
+
+  if (route.GetTraffic().empty())
+  {
+    CHECK_GREATER_OR_EQUAL(route.m_poly.GetPolyline().GetSize(), 1, ());
+    m_traffic.insert(m_traffic.end(),
+                     route.m_poly.GetPolyline().GetSize() - 1 /* segment number is less by one */,
```

Comment:
Not a "segment number". "Segment number" is a number of a particular segment. Please, use "number of segments".